enterprise/server/backends/userdb/userdb.go

```diff
@@ -535,7 +535,7 @@ func (d *UserDB) GetGroupUsers(ctx context.Context, groupID string, opts *interf
 		Role       role.Role
 	}
 
-	if authutil.UserListsEnabled() {
+	if authutil.UserListsEnabled() && (slices.Contains(opts.Statuses, grpb.GroupMembershipStatus_MEMBER) && opts.SubIDPrefix == "") {
```

Comment:
nit: could you add a comment explaining this logic?

Reply:
done